experimental/ActionPolyRing/src/Content.jl

```diff
@@ -0,0 +1,1162 @@
+                 index_ordering_name::Symbol = :default,
+                 partition::Vector{Vector{Int}} = Vector{Int}[],
+                 index_ordering_matrix::ZZMatrix = zero_matrix(ZZ, 0, 0)) 
+
+This method configures the ranking of the action polynomial ring `apr`, using an ordered partition of the elementary symbols and a monomial ordering on the indices. The ranking can be specified either by choosing predefined naming options or by explicitly providing a custom configuration.
+
+# Keyword Arguments
+- `partition_name`: Determines the partition of the elementary symbols of `dpr`. Supported values are:
+  - `:top`: groups all variables into a single block,
+  - `:pot`: separates each variable into its own block,
+  - `:default`: uses `:top` unless a custom partition is specified.
+
+- `index_ordering_name`: Specifies the ordering on the multiindices. Supported values are:
+  - `:lex`: lexicographic ordering,
+  - `:deglex`: degree lexicographic ordering,
+  - `:invlex`: inverse lexicographic ordering,
+  - `:deginvlex`: degree inverse lexicographic ordering,
+  - `:degrevlex`: degree reverse lexicographic ordering,
+  - `:default`: uses `:lex` unless a custom matrix is specified.
+
+- `partition`: A custom partition of the elementary symbols, represented as a vector of characteristic vectors. The elementary symbols corresponding to the first characteristic vectors are considered largest and so on.
+
+- `index_ordering_matrix`: A custom matrix representing a monomial ordering on the indices. Its number of columns must equal `ndiffs(dpr)`.
+
+# Examples
+
+```jldoctests
+julia> dpr = differential_polynomial_ring(ZZ, [:a, :b, :c], 4; partition_name=:pot, index_ordering_name = :degrevlex)[1]; ranking(dpr)
+Ranking of differential polynomial ring in 3 elementary symbols over ZZ
+with elementary symbols partitioned by
+  [[1, 0, 0], [0, 1, 0], [0, 0, 1]]
+and ordering of the indices defined by
+  [1    1    1    1]
+  [0    0    0   -1]
+  [0    0   -1    0]
+  [0   -1    0    0]
+
+julia> set_ranking!(dpr; partition = [[0,1,1],[1,0,0]], index_ordering_matrix = identity_matrix(ZZ, 4))
+Ranking of differential polynomial ring in 3 elementary symbols over ZZ
+with elementary symbols partitioned by
+  [[0, 1, 1], [1, 0, 0]]
+and ordering of the indices defined by
+  [1   0   0   0]
+  [0   1   0   0]
+  [0   0   1   0]
+  [0   0   0   1]
+
+julia> set_ranking!(dpr)
+Ranking of differential polynomial ring in 3 elementary symbols over ZZ
+with elementary symbols partitioned by
+  [[1, 1, 1]]
+and ordering of the indices defined by
+  [1   0   0   0]
+  [0   1   0   0]
+  [0   0   1   0]
+  [0   0   0   1]
+```
+"""
+function set_ranking!(dpr::PolyT;
+    partition_name::Symbol = :default,
+    index_ordering_name::Symbol = :default,
+    partition::Vector{Vector{Int}} = Vector{Int}[],
+    index_ordering_matrix::ZZMatrix = zero_matrix(ZZ, 0, 0)
+  )::ActionPolyRingRanking{PolyT} where {PolyT <: ActionPolyRing}
```

Comment:
```suggestion
  )
```
in julia, we don't annotate return types, as that leads to the compiler adding hidden `convert` statements into the compiled code which can lead to hard-to-find bugs